lib/stargate.ex

```diff
@@ -82,6 +82,23 @@ defmodule Stargate do
   defdelegate produce(url_or_connection, message), to: Stargate.Producer
   defdelegate produce(connection, message, mfa), to: Stargate.Producer
 
+  @type tenant :: String.t()
+  @type namespace :: String.t()
+  @type topic :: String.t()
+  @type persistence :: String.t()
+  @type component :: :producer | :producer_ack | :consumer | :consumer_ack | :reader | :reader_ack
+  @type key_opt :: {:persistence, persistence()} | {:name, atom()} | {:component, component()}
+
+  @spec registry_key(tenant(), namespace(), topic(), [key_opt]) ::
+          {:via, Registry, {atom(), {component(), persistence(), tenant(), namespace(), topic()}}}
+  def registry_key(tenant, namespace, topic, opts \\ []) do
+    name = Keyword.get(opts, :name, :default)
+    component = Keyword.get(opts, :component, :producer)
+    persistence = Keyword.get(opts, :persistence, "persistent")
+
+    {:via, Registry, {:"sg_reg_#{name}", {component, persistence, tenant, namespace, topic}}}
```

Comment:
I think this should be

```suggestion
    {:via, Registry, {name, {component, persistence, tenant, namespace, topic}}}
```
Since I can currently do:
```
Stargate.Producer.start_link([registry: MyProject.Stargate.Registry])
Stargate.Producer.produce(
           {:via, Registry, {MyProject.Stargate.Registry, registry_key}},
           message
)
```

Reply:
This wouldn't really work as is for anyone using the top-level supervisor to manage the entire stack, since this allows you to pick a unique name for the entire supervision tree and then wraps the named components (registry and supervisor) in `sg_reg` and `sg_sup` prefixes to prevent them from colliding but I've modified this function to allow you to supply the name of your supervision stack _or_ the `:registry` option if you're using your own registry